x-pack/plugins/ingest_manager/scripts/dev_agent/script.ts

```diff
@@ -111,24 +111,23 @@ async function enroll(kibanaURL: string, apiKey: string, log: ToolingLog): Promi
       },
     },
   };
-  const res = await fetch(`${kibanaURL}/api/ingest_manager/fleet/agents/enroll`, {
-    method: 'POST',
-    body: JSON.stringify(body),
-    headers: {
-      'kbn-xsrf': 'xxx',
-      Authorization: `ApiKey ${apiKey}`,
-      'Content-Type': 'application/json',
-    },
-  });
-  const obj: PostAgentEnrollResponse = await res.json();
-
-  if (!obj.success) {
-    log.error(JSON.stringify(obj, null, 2));
+  try {
+    const res = await fetch(`${kibanaURL}/api/ingest_manager/fleet/agents/enroll`, {
+      method: 'POST',
+      body: JSON.stringify(body),
+      headers: {
+        'kbn-xsrf': 'xxx',
+        Authorization: `ApiKey ${apiKey}`,
+        'Content-Type': 'application/json',
+      },
+    });
+    const obj: PostAgentEnrollResponse = await res.json();
+    return {
```

Comment:
Quick Question - I think I have to update an external CLI tool and a draft PR to remove relying on this `success` flag. In this case here, what would throw? because I thought that in a condition where the server returns a non-200 code, `res.json()` would actually parse ok.
Do you need to do a `res.ok()` and then manually `throw`?

Reply:
@paul-tavares Nice one. Thanks! Fresh :eyes: and work on https://github.com/elastic/kibana/pull/74507 makes me think the same.

I think I'll revert the `try/catch` wrap and do something like

```diff
- if (!obj.success) {
+ if (!res.ok()) {
```

Reply:
yeah, that will work 👍